### ISSUE TITLE
Can't set tax back to zero after setting it to non-zero value.

### ISSUE THREAD

**Issue by ssoroka** [stripe/stripe-go#225]
Seeing as how TaxPercent isn't set in the request if it's zero, there's no way to actually set it to zero. This is a problem if your tax rate has been set previously and you want to change it.

**Comment by brandur** [comment]
So just to clarify this a bit, the problem is related to parameters on both subscriptions and invoices and arises because of the "> 0" check that's performed when serializing parameters:

``` go
if params.TaxPercent > 0 {
	body.Add("tax_percent", strconv.FormatFloat(params.TaxPercent, 'f', 2, 64))
}
```

*I think* we can probably just resolve this by removing the check on serialization, but this obviously has the slightly undesirable effect of having the `tax_percent` with every create or update request (although that shouldn't be harmful).

Grepping through the codebase though, it seems that this applies beyond just `tax_percent` to a number of other fields that are using the "> 0" convention; all of which could be prone to a similar bug. @cosn Was there any original thinking around this design which I'm missing and which was supposed to compensate for this problem?

**Comment by ssoroka** [comment]
I think setting tax on every request is harmful because that then requires the developer to know and resend it every request. Omitting the value is then the same as setting it to zero.

The other fields seem to solve this by having a second parameter that can be set to true if you want a zero. See `QuantityZero`. So we'd need something like `TaxPercentZero`

**Comment by brandur** [comment]
> I think setting tax on every request is harmful because that then requires the developer to know and resend it every request. Omitting the value is then the same as setting it to zero.

Oh, yep. Absolutely right. Thanks!

> The other fields seem to solve this by having a second parameter that can be set to true if you want a zero. See QuantityZero. So we'd need something like TaxPercentZero

True! It's definitely a little janky, but seems to be the only option that will fit into the current framework. I've created a pull with the change in #226.

**Comment by brandur** [comment]
Patch and released as 13.9.0.

**Comment by cosn** [comment]
yup, the reason mentioned by @ssoroka is exactly why we have that check (and all the others). The boolean-based pattern, while not great, will indeed work.
